### PR TITLE
feat: new backend `k8s` and refactor state options validation

### LINKED ISSUE #995: :four_leaf_clover: `Proposal`: Add k8s as a Backend of the State
## What Would You Like to Add? Why Is This Needed?

We have supported for `Local file` and `S3` as the `State` Backends, I believe it is a useful feature to support `k8s` as a `Backend` in an offline k8s environment.

## Design

Save `State` via `ConfigMap`.

## Anything else

Who'd like to do it? @aFlyBird0 @iyear

### PR DESCRIPTION
**Pull request by aFlyBird0**
Signed-off-by: Bird <aflybird0@gmail.com>

## Pre-Checklist

Note: please complete **_ALL_** items in the following checklist.

- [x] I have read through the [CONTRIBUTING.md](https://github.com/devstream-io/devstream/blob/main/CONTRIBUTING.md) documentation.
- [x] My code has the necessary comments and documentation (if needed).
- [x] I have added relevant tests

## Description
* Use K8s ConfigMap as a backend.
* Validate backend options in pkg `backend` during calling `GetBackend` to unify and remove duplicate validations.

## Related Issues
close #995 

## New Behavior (screenshots if needed)

image

image

image

### CODE REVIEW

File: internal/pkg/backend/types/vars.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+func (t Type) String() string {
+	return string(t)
+}
+
+const (
+	Local   Type = "local"
+	S3      Type = "s3"
+	K8s     Type = "k8s"
+	K8sAlis Type = "kubernetes"
```
Comment (aFlyBird0) [inline internal/pkg/backend/types/vars.go:13]: alias

**Review by IronCore864 — APPROVED** [review]

File: internal/pkg/backend/backend.go [hunk 2]
```diff
@@ -1,22 +1,16 @@
...
 
-type Type string
-
-const (
-	Local Type = "local"
-	S3    Type = "s3"
-)
-
-// Backend is used to persist data, it can be local file/etcd/s3/...
+// Backend is used to persist data, it can be local file/etcd/s3/k8s...
```
Comment (daniel-hutao) [inline internal/pkg/backend/backend.go:13]: no etcd

File: docs/core-concepts/stateconfig.md [hunk 3]
```diff
@@ -22,6 +23,20 @@ state:
...
+```yaml
+varFile: variables-gitops.yaml
+
+toolFile: tools-gitops.yaml
+
+state:
+  backend: k8s
+  options:
+    namespace: devstream        # optional, default is "devstream", will be created if not exists
+    configmap: devstream-state  # optional, default is "devstream-state", will be created if not exists
```
Comment (daniel-hutao) [inline docs/core-concepts/stateconfig.md:37]: How about to use ConfigMap name "state"? devstream/state vs devstream/devstream-state

**Review by daniel-hutao — CHANGES_REQUESTED** [review]

**Review by iyear — APPROVED** [review]
LGTM

**Comment by aFlyBird0** [comment]
Wait for #1016 to be merged.

**Review by daniel-hutao — APPROVED** [review]

**Pull request merged**